### PR TITLE
[CodingStyle] Re-use assigned variable for Method/Static/FuncCall/New_ on SplitDoubleAssignRector

### LINKED ISSUE #4948: Rector\CodingStyle\Rector\Assign\SplitDoubleAssignRector: doubles call to a function
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | dev-master              |
| Installed as   | composer dependency                  |

It doubles function call. That can have unexpected consequences.

## Minimal PHP Code Causing Issue

https://getrector.org/demo/46512990-cd60-4fdc-a22f-9cc2da52ff23#demo_result

## Expected Behaviour

```php
$dlpayload = $payload->DataLives = get_whatever();
```

sould become:

```php
$dlpayload = get_whatever();
$payload->DataLives = $dlpayload;
```

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes #4948

### CODE REVIEW

**Comment by samsonasik** [comment]
also ensure parent assign is an Expression statement

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: rules/coding-style/tests/Rector/Assign/SplitDoubleAssignRector/Fixture/skip_assign_in_non_expression.php.inc [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace Rector\CodingStyle\Tests\Rector\Assign\SplitDoubleAssignRector\Fixture;
+
+class SkipAssignInNonExpression
+{
+    public function run()
+    {
+        if ($one = $two = 1) {
+
+        }
```
Comment (TomasVotruba) [inline rules/coding-style/tests/Rector/Assign/SplitDoubleAssignRector/Fixture/skip_assign_in_non_expression.php.inc:11]: :+1:

**Review by TomasVotruba — APPROVED** [review]
Very nice

**Comment by TomasVotruba** [comment]
Thanks!

**Pull request merged**